src/rgw/rgw_amqp.cc

```diff
@@ -0,0 +1,240 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+
+//#include "rgw_amqp.h"
+#include <amqp.h>
+#include <amqp_tcp_socket.h>
+
+#include <sstream>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+
+// comparisson operator needed so that amqp_connection_info could
+// be used as a key in std::map
+bool operator<(const amqp_connection_info& lhs, const amqp_connection_info& rhs) {
+  // check hostname
+  if (std::strcmp(lhs.host, rhs.host) < 0) return true;
+  if (std::strcmp(lhs.host, rhs.host) > 0) return false;
+  // same hostname, check port
+  if (lhs.port < rhs.port) return true;
+  if (lhs.port > rhs.port) return false;
+  // same port, check vhost
+  if (std::strcmp(lhs.vhost, rhs.vhost) < 0) return true;
+  if (std::strcmp(lhs.vhost, rhs.vhost) > 0) return false;
+  // both are equal (ignoring user/password and ssl support)
+  return false;
+}
+
+namespace amqp {
+
+// convert connection ifo to string
+std::string to_str(const amqp_connection_info& info) {
+  std::stringstream ss;
```

Comment:
would prefer the new fast fmt mechanism now in the codebase, adam emerson has some examples, I think?

Reply:
will look into that